src/sagemaker/estimator.py

```diff
@@ -833,6 +875,82 @@ def __init__(self, entry_point, source_dir=None, hyperparameters=None, enable_cl
+        except subprocess.CalledProcessError:
+            raise subprocess.CalledProcessError(1, cmd='git clone {} {}'.format(self.git_config['repo'], repo_dir))
+
+        self._checkout_branch_and_commit(repo_dir)
+
+        # check if the cloned repo contains entry point and source dir; if so, set ``entry_point`` and
+        # ``source_dir`` to the paths to local file system.
+        if self.source_dir:
+            if os.path.isdir(os.path.join(repo_dir, self.source_dir)):
+                self.source_dir = os.path.join(repo_dir, self.source_dir)
+                os.chdir(self.source_dir)
+            else:
+                raise ValueError('Source directory does not exist in the repo.')
+            if not os.path.isfile(os.path.join(self.source_dir, self.entry_point)):
+                raise ValueError('Entry point does not exist in the repo.')
+        else:
+            if not os.path.isfile(os.path.join(repo_dir, self.entry_point)):
+                raise ValueError('Entry point does not exist in the repo.')
+        for path in self.dependencies:
+            if not os.path.isdir(os.path.join(repo_dir, path)):
+                raise ValueError('Dependency {} does not exist in the repo.'.format(path))
+
+    def _checkout_branch_and_commit(self, repo_dir):
+        """Enter the directory where the repo is cloned, and  checkout the required branch and commit.
+
+        Args:
+            repo_dir: the directory where the repo is cloned
+
+        Raises:
+            ValueError: If 1. entry point specified does not exist in the repo
+                           2. source dir specified does not exist in the repo
+        """
+        os.chdir(repo_dir)
+        if 'branch' in self.git_config:
+            try:
+                subprocess.check_call(['git', 'checkout', self.git_config['branch']])
+            except subprocess.CalledProcessError:
+                raise subprocess.CalledProcessError(1, cmd='git checkout {}'.format(self.git_config['branch']))
+        if 'commit' in self.git_config:
+            try:
+                subprocess.check_call(['git', 'checkout', self.git_config['commit']])
+            except subprocess.CalledProcessError:
+                raise subprocess.CalledProcessError(1, cmd='git checkout {}'.format(self.git_config['commit']))
+
+    def _validate_git_config(self):
+        """check if a git_config param is valid
+
+        Raises:
+            ValueError: If:
+                1. git_config has no key 'repo'
+                2. git_config['repo'] is in the wrong format.
+        """
+        if 'repo' not in self.git_config:
+            raise ValueError('Please provide a repo for git_config.')
+        repo = self.git_config['repo']
+        codecommit_url = repo.startswith('https://git-codecommit') or repo.startswith('ssh://git-codecommit')
+        github_url = repo.startswith('https://github') or repo.startswith('git@github')
```

Comment:
No reason to limit this feature only to git hub and code commit. Let's remove this validation here and allow customer to use any git repository.